iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/SubscriptionOwnerCapabilitiesCheck.java

```diff
@@ -0,0 +1,87 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.iac.arm.checkdsl.ContextualObject;
+import org.sonar.iac.arm.checkdsl.ContextualResource;
+import org.sonar.iac.arm.tree.api.Expression;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.api.tree.Tree;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.sonar.iac.common.checks.TextUtils.isValue;
+import static org.sonar.iac.common.checks.TextUtils.matchesValue;
+
+@Rule(key = "S6385")
+public class SubscriptionOwnerCapabilitiesCheck extends AbstractArmResourceCheck {
+  private static final String MESSAGE = "Narrow the number of actions or the assignable scope of this custom role.";
+  private static final String PERMISSION_MESSAGE = "Allows all actions.";
+  private static final String SCOPE_MESSAGE = "High scope level.";
+
+  private static final Pattern RESOURCE_GROUP_SCOPE_PATTERN = Pattern.compile("/subscriptions/[^/]+/resourceGroups/.*");
+
+  @Override
+  protected void registerResourceConsumer() {
+    register("Microsoft.Authorization/roleDefinitions", SubscriptionOwnerCapabilitiesCheck::checkSubscriptionOwnerCapabilities);
+  }
+
+  private static void checkSubscriptionOwnerCapabilities(ContextualResource resource) {
+    Optional<Expression> arbitraryAction = resource.list("permissions")
+      .objects()
+      .flatMap(SubscriptionOwnerCapabilitiesCheck::findArbitraryActions)
+      .findAny();
+
+    if (arbitraryAction.isEmpty()) {
+      return;
+    }
+
+    List<Tree> sensitiveScopes = resource.list("assignableScopes")
+      .getItemIf(SubscriptionOwnerCapabilitiesCheck::isSensitiveScope)
+      .collect(Collectors.toList());
+
+    if (sensitiveScopes.isEmpty()) {
+      return;
+    }
+
+    List<SecondaryLocation> secondaries = sensitiveScopes.stream().map(it -> new SecondaryLocation(it, SCOPE_MESSAGE)).collect(Collectors.toList());
+    secondaries.add(0, new SecondaryLocation(arbitraryAction.get(), PERMISSION_MESSAGE));
+    resource.report(MESSAGE, secondaries);
+  }
+
+  private static Stream<Expression> findArbitraryActions(ContextualObject permissionObject) {
+    return permissionObject.list("actions").getItemIf(
+      action -> isValue(action, "*").isTrue());
+  }
+
+  private static boolean isSensitiveScope(Tree scope) {
+    boolean hasSensitiveCall = isValue(scope, "[managementGroup().id]").isTrue() ||
+      isValue(scope, "[subscription().id]").isTrue();
+    boolean referencesSensitiveScope = matchesValue(scope, s -> s.startsWith("/subscriptions/")).isTrue() &&
+      matchesValue(scope, s -> RESOURCE_GROUP_SCOPE_PATTERN.matcher(s).matches()).isFalse() ||
```

Comment:
I would use the same approach as for the terraform rule, with this regex: `^/subscriptions/[^/]+/?$`
That way you don't need to perform two checks on the same string.